### DISCUSSION TITLE
Threads in Pluto

### DISCUSSION THREAD

**Discussion by r-acad** [fonsp/Pluto.jl#1175] [Questions about using Pluto] [answered]
I start Julia from the command line with ***julia -t 32*** and I see 32 threads (***Threads.nthreads()*** returns 32 in the REPL)

I then start Pluto and I only see 16 (***Threads.nthreads()*** returns 16 when run in a Pluto cell)

Then I solve a linear system in the REPL (***A\b***, where A is a dense matrix and b a dense vector) and I see all the cores running in parallel in the resource monitor (Win10), but when I do this within Pluto I can only see one core working in this case.

In my code some sections seem to run in parallel and some others (which should be multithreaded) run on a single core.

**Is this a limitation of Pluto or I am doing something wrong?** 

I am running Julia 1.6.1 and Pluto 0.14.5

Raul

**Comment by fonsp** [comment, marked as answer]
The Pluto server starts a new terminal for each notebook, and you specify the number of threads for _notebook processes_ with:
```julia
Pluto.run(threads=16)
```

Using `julia -t 32` to run Pluto does not work because it sets the number of threads for the Pluto _server_, not the processes that the server creates.

**Reply by r-acad** [reply]
Fons, you are my hero 🙏

**Reply by lungben** [reply]
Even without explicitly setting `threads`, Pluto uses a sensible default value (I think it is logical cores / 2 to account for hyperthreading - using more threads than physical cores usually gives a slow-down).
